packages/playwright-test/src/matchers/matchers.ts

```diff
@@ -252,14 +265,12 @@ export function toHaveURL(
   }, expected, options);
 }
 
-export function toHaveValue(
+export async function toHaveScreenshot(
   this: ReturnType<Expect['getState']>,
-  locator: LocatorEx,
-  expected: string | RegExp,
-  options?: { timeout?: number },
+  page: Page,
+  nameOrOptions: Parameters<typeof toMatchSnapshot>[1],
+  optOptions: Parameters<typeof toMatchSnapshot>[2],
 ) {
-  return toMatchText.call(this, 'toHaveValue', locator, 'Locator', async (isNot, timeout) => {
-    const expectedText = toExpectedTextValues([expected]);
-    return await locator._expect('to.have.value', { expectedText, isNot, timeout });
-  }, expected, options);
+  const screenshot = await page.screenshot();
```

Comment:
shall we pick screenshot type using screenshot name extension?

Reply:
let maybe not do it for now, since we go into the nameless snapshot world anyway?